apps/server/src/modules/tool/tool-launch/service/auto-parameter-strategy/auto-group-external-uuid-strategy.service.ts

```diff
@@ -0,0 +1,51 @@
+import { Injectable } from '@nestjs/common';
+import { CourseService } from '@modules/learnroom';
+import { Group, GroupService } from '@modules/group';
+import { Course } from '@shared/domain/entity';
+import { CustomParameterType, ToolContextType } from '../../../common/enum';
+import { ContextExternalToolLaunchable } from '../../../context-external-tool/domain';
+import { SchoolExternalTool } from '../../../school-external-tool/domain';
+import { MissingAutoParameterValueLoggableException } from '../../error';
+import { AutoParameterStrategy } from './auto-parameter.strategy';
+
+@Injectable()
+export class AutoGroupExternalUuidStrategy implements AutoParameterStrategy {
+	constructor(private readonly courseService: CourseService, private readonly groupService: GroupService) {}
+
+	async getValue(
+		_schoolExternalTool: SchoolExternalTool,
+		contextExternalTool: ContextExternalToolLaunchable
+	): Promise<string | undefined> {
+		if (contextExternalTool.contextRef.type !== ToolContextType.COURSE) {
+			return undefined;
+		}
+
+		const courseId = contextExternalTool.contextRef.id;
+		const course: Course = await this.courseService.findById(courseId);
+
+		const syncedGroup: Group | undefined = await this.getSyncedGroup(course);
+		if (!syncedGroup) {
+			return undefined;
+		}
+
+		const groupUuid = syncedGroup.externalSource?.externalId;
+		if (!groupUuid) {
+			throw new MissingAutoParameterValueLoggableException(
```

Comment:
I think we should not throw an error, since this also happens when the parameter is optional. It is unlikely that a course is synchronized with an internal group, but if this ever happens it would break.